src/main/kotlin/domain/Matrix.kt

```diff
@@ -10,16 +10,18 @@ class Matrix(
     init {
         val locations = MatrixFiller.fill(dimension)
         val miningLocations = locationSelector.select(numberOfMines.mineCount, locations)
-        cells = locations.map {
-            when (it) {
-                in miningLocations -> Cell.mine(it)
-                else -> Cell.ground(it)
-            }
-        }.associateBy {
+        cells = locations.map { getCell(it, miningLocations) }.associateBy {
             it.location
         }.toMap()
     }
 
+    private fun getCell(location: Location, miningLocations: List<Location>): Cell {
+        return when (location) {
+            in miningLocations -> Cell.mine(location)
+            else -> Cell.ground(location)
+        }
```

Comment:
```suggestion
        return if (location in miningLocations) {
            Cell.mine(location)
        } else {
            Cell.ground(location)
        }
```
코틀린 코딩 컨벤션에서는 3개 이상일 때부터 when을 쓰도록 권장하고 있습니다 : )

참고
[ifelse vs when](https://kotlinlang.org/docs/coding-conventions.html#if-versus-when)